FaunaDB.Client/Query/Language.Miscellaneous.cs

```diff
@@ -920,5 +920,121 @@ public partial struct Language
         /// </summary>
         public static Expr Documents(Expr collection) =>
             UnescapedObject.With("documents", collection);
+        
+        /// <summary>
+        /// Try to convert an object into an array of (field, value).
+        /// <para>
+        /// See the <see href="https://docs.fauna.com/fauna/current/api/fql/functions/toarray">ToArray</see>
+        /// </para>
+        /// </summary>
+        public static Expr ToArray(Expr expr) =>
+            UnescapedObject.With("to_array", expr);
+        
+        /// <summary>
+        /// Try to convert an array of (field, value) into an object.
+        /// <para>
+        /// See the <see href="https://docs.fauna.com/fauna/current/api/fql/functions/toobject">ToObject</see>
+        /// </para>
+        /// </summary>
+        public static Expr ToObject(Expr fields) =>
+            UnescapedObject.With("to_object", fields);
+        
+        /// <summary>
+        /// Casts an expression to a double value, if possible.
+        /// <para>
+        /// See the <see href="https://docs.fauna.com/fauna/current/api/fql/functions/todouble">ToDouble</see>
+        /// </para>
+        /// </summary>
+        public static Expr ToDouble(Expr value) =>
+            UnescapedObject.With("to_double", value);
+        
+        /// <summary>
+        /// Casts an expression to an integer value, if possible.
+        /// <para>
+        /// See the <see href="https://docs.fauna.com/fauna/current/api/fql/functions/tointeger">ToInteger</see>
+        /// </para>
+        /// </summary>
+        public static Expr ToInteger(Expr value) =>
+            UnescapedObject.With("to_integer", value);
+
+        /// <summary>
+        /// Returns a new time or date with the offset in terms of the unit added.
+        /// <para>
+        /// See the <see href="https://docs.fauna.com/fauna/current/api/fql/functions/timeadd">TimeAdd</see>
+        /// </para>
+        /// </summary>
+        public static Expr TimeAdd(Expr value, Expr offset, Expr unit) =>
+            UnescapedObject.With("time_add", value, "offset", offset, "unit", unit);
+        
+        /// <summary>
+        /// Returns a new time or date with the offset in terms of the unit added.
+        /// <para>
+        /// See the <see href="https://docs.fauna.com/fauna/current/api/fql/functions/timeadd">TimeAdd</see>
+        /// </para>
+        /// </summary>
+        public static Expr TimeAdd(Expr value, Expr offset, FaunaTimeUnit unit) =>
+            TimeAdd(value, offset, unit.Value);
+        
```

Comment:
I think it's  nice to wrap unit as a new type `FaunaTimeUnit`, but, can't we use an `enumerate` here ?

Reply:
thanks a lot for noticing this one :)
I did a bad thing by introducing that new type.